app/controllers/Team.scala

```diff
@@ -139,24 +141,58 @@ final class Team(
       }
     }
 
-  def kickForm(id: String) =
+  def manageMembersForm(id: String) =
     Auth { implicit ctx => me =>
       WithOwnedTeamEnabled(id) { team =>
-        env.team.memberRepo userIdsByTeam team.id map { userIds =>
-          html.team.admin.kick(team, userIds.filter(me.id !=))
-        }
+        for {
+          userIds <- env.team.memberRepo userIdsByTeam team.id
+          classes <- isGranted(_.Teacher) ?? {
+            clas.clas
+              .of(me)
+              .flatMap(_.map { myClas =>
+                clas.student.activeOf(myClas).map { students =>
+                  myClas.withStudents(
+                    students.filter(student => student.managed && !userIds.contains(student.userId))
+                  )
+                }
+              }.sequenceFu)
+          }
+        } yield html.team.admin.members(
+          team,
+          forms.manageMembers,
+          userIds.filter(me.id !=),
+          classes.filter(_.students.nonEmpty)
+        )
       }
     }
 
-  def kick(id: String) =
+  def manageMembers(id: String) =
     AuthBody { implicit ctx => me =>
       WithOwnedTeamEnabled(id) { team =>
         implicit val req = ctx.body
-        forms.selectMember.bindFromRequest().value ?? { api.kick(team, _, me) } inject Redirect(
-          routes.Team.kickForm(team.id)
-        ).flashSuccess
+        forms.manageMembers.bindFromRequest().value ?? { case (students, members) =>
+          students
+            .split(",")
+            .toList
+            .map { userId =>
+              userRepo
+                .byId(userId)
+                .flatMap(_ ?? { user =>
+                  api.doJoin(team, user)
```

Comment:
this allows adding any arbitrary users to a team, students or not.

Reply:
check has been added to allow only managed students of logged in teacher to add new team members